### PR TITLE
Introduce Test 1-24 functionality

### PR DESCRIPTION
**Pull request by SalDaniele**
This commit includes some general refactoring to aid with organizing the various configurations and adds some logic to set reasonable defaults based on the test case ids.

With these changes discrete trials of tests 1 - 24 should work.

Still TODO to achieve basic parity with legacy FT tool:
- Pod to External Testing
- HWOL detection monitor
- Add basic driver to run program from user config rather than hardcoded values

### CODE REVIEW

File: testSettings.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+            Index={self.client_index}
+        Server Node: {self.node_server_name}
+            Exec Persistence: {self.server_is_persistent}
+            Tenant={self.server_is_tenant}
+            Index={self.server_index}
+        """
+
+    def _test_id_to_connection_mode(self, test_case_id) -> ConnectionMode:
+        """The connection type will be used to determine what IP the client should direct traffic to"""
+        if test_case_id in (5, 6, 7, 8, 17, 18, 19, 20):
```
Comment (wizhaoredhat) [inline testSettings.py:39]: Maybe we can have a singular function that sets all the variables in this class E.g.:
```
def get_all_params_from_test_id(self, test_case_id) 
        self.service = ConnectionMode.POD_ID
        if test_case_id in (9, 10, 11, 12, 21, 22, 23, 24):
            self.service =  ConnectionMode.NODE_PORT_IP
        self.nodeLocation = NodeLocation.DIFF_NODE
        if test_case_id in (1, 3, 5, 7, 9, 11, 13, 15, 17, 19, 21, 23):
              self.nodeLocation = NodeLocation.SAME_NODE
```

File: testSettings.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
+from common import TestCaseType, PodType, ConnectionMode
+
+class TestSettings():
```
Comment (wizhaoredhat) [inline testSettings.py:3]: I like this new Class!

File: trafficFlowTests.py [hunk 3]
```diff
@@ -197,29 +109,43 @@ def run(self):
...
+            node_client_name = "worker-235"
+            index = 0
+            pod_type = "sriov"
+            ###################
+
+            self.test_settings = TestSettings(
+                test_case_id=test_id,
+                node_server_name=node_server_name,
+                node_client_name=node_client_name,
+                server_pod_type=self.server_test_to_pod_type(test_id, pod_type),
```
Comment (wizhaoredhat) [inline trafficFlowTests.py:133]: I think we can move this function into TestSetting what do you think?

**Review by wizhaoredhat — COMMENTED** [review]
Overall I like the changes! Just some minor comments.

File: iperf.py [hunk 4]
```diff
@@ -35,14 +36,14 @@ def __init__(self, tft: TestConfig, index: int, node_name: str, exec_persistent:
...
-            self.template_args["args"] = f"-s -p {self.port}"
+            self.template_args["command"] = "iperf3"
+            self.template_args["args"] = ["-s", "-p", f"{self.port}"]
 
         common.j2_render(self.in_file_template, self.out_file_yaml, self.template_args)
         logger.info(f"Generated Server Pod Yaml {self.out_file_yaml}")
 
-        self.create_cluster_ip_service()
-        self.create_node_port_service(self.port + 25000)
+        self.cluster_ip_addr = self.create_cluster_ip_service()
```
[the remark this answers is not in the document]
Comment (SalDaniele) [reply iperf.py:45]: Yes I agree, I'll make a note to add that in a future PR

File: trafficFlowTests.py (hunk 3 above)
Comment (wizhaoredhat) [reply trafficFlowTests.py]: Looks good now

**Comment by wizhaoredhat** [comment]
LGTM

**Pull request merged**